libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -749,10 +746,14 @@ class MapboxRouteLineApi(
                         MapboxRouteLineUtils.getRestrictedLineExpression(
                             offset,
                             -1,
-                            routeLineOptions
+                            restrictedSectionColor = routeLineOptions
                                 .resourceProvider
                                 .routeLineColorResources
                                 .restrictedRoadColor,
+                            restrictedSectionInactiveColor = routeLineOptions
```

Comment:
Don't you also have to check here that independent styling is enabled for inactive legs? You do that when you invoke `getRestrictedLineExpression` from another place.

Reply:
Good catch!

Reply:
Continuing in https://github.com/mapbox/mapbox-navigation-android/pull/7328.

Reply:
I meant that you have to check styleInactiveLegsIndependently flag to decide whether to use inactive restricted colour or the usual one. But yeah, activeLegIndex was also not taken into account.

Reply:
Yea, even when we check `styleInactiveLegsIndependently` it's irrelevant if leg index is hardcoded to `-1`.